packages/components/src/font-size-picker/utils.js

```diff
@@ -22,12 +22,13 @@ const CUSTOM_FONT_SIZE_OPTION = {
  * @return {[number, string]} An array with the numeric value and the unit if exists.
  */
 export function splitValueAndUnitFromSize( size ) {
-	/**
-	 * The first matched result is ignored as it's the left
-	 * hand side of the capturing group in the regex.
-	 */
-	const [ , numericValue, unit ] = size.split( /(\d+)/ );
-	return [ numericValue, unit ];
+	const [ numericValue, unit ] = `${ size }`.match( /[\d\.]+|\D+/g );
```

Comment:
Looking at the changes made in #36420, I believe that we should also apply the change on the JSDoc for the `size` param (i.e. changing its type to be only `string`)

Reply:
The changes here were slightly different to #36420. It honours what was documented in terms of the param accepting `string|number`. The tests also cover both strings and numbers.

The `FontSizePicker` [docs](https://github.com/WordPress/gutenberg/blob/21831a97dd0e1cba21678acd047e44b71b6bd037/packages/components/src/font-size-picker/README.md) themselves also include numerical `size` values. Should we not maintain compatibility with that?

In light of that, do you still believe I should restrict this?

Reply:
You actually made a few good points — let's leave it as-is (and thank you for looking into it)

Reply:
I was wondering why you changed from `split` to `match`. I was expecting a change in the regex.

Reply:
Honestly, this wasn't a conscious choice. It came from when I copied the [original change](https://github.com/WordPress/gutenberg/pull/36420/files#diff-e889984e764a872b1af0abd101bce70d90341b301b84aa9f7322d979465acf3fR25). 

On a closer look though, `match` appears a little cleaner in terms of its results. I don't feel strongly about it, so if there is a good reason to change it back to `split` I can.